### DISCUSSION TITLE
Too many DaisyUI logs in dev mode.

### DISCUSSION THREAD

**Discussion by onpix** [saadeghi/daisyui#1391] [Q&A] [answered]
I use daisyUI and Nuxt3. I found that there are too many logs when I'm in dev mode (using vite, HMR enabled):

![Imgur](https://i.imgur.com/gzZwGpe.png)

It seems that a new log is printed each time I change my source file (even though I did not modify daisyUI classes). And the webpage update delay is obvious. I guess maybe this is because daisyUI package is initialized each time I change source files and as a result, the HMR delay becomes larger? Is there any method to solve this? Thanks in advance.

**Comment by saadeghi** [comment]
It should only print once.  
[Here's a Vite example](https://stackblitz.com/edit/daisyui-vite?file=src%2Fmain.js) and you can see it only logs once on the first run. even changing the class names does not print a log again.  
So I think that might be a problem from either your build tool or your config. Maybe it's running the build process on every change.

Can you please share a small GitHub repo with me so I can reproduce the issue?

Also, you can simply disable the daisyUI logs if you don't need it:  
https://daisyui.com/docs/config/
```js
module.exports = {
  //...
  daisyui: {
    logs: false,
  },
}
```

**Reply by onpix** [reply]
I found it! It seems that this is caused by the tailwind CSS directives in my code (in `.css` file or `.vue` file), 
but I'm not sure whether it is caused by daisyUI or tailwindcss or nuxtjs.

I make a small demo here: [https://stackblitz.com/edit/github-6auzzw?file=assets/main.css](https://stackblitz.com/edit/github-6auzzw?file=assets/main.css)

By removing the following code, the duplicated logs will disappear.

```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

**Comment by saadeghi** [comment, marked as answer]
I'm not sure how Nuxt is processing the CSS but for some reason, when you have your CSS file in `assets/main.css` and you're importing it to `nuxt.config.ts` it forces a fresh rebuild on every change.  

Instead, rename your CSS to `/assets/css/tailwind.css` [as the tutorial says](https://tailwindcss.nuxt.dev/getting-started/setup/). This way you don't even need to import it to `nuxt.config.ts` and it doesn't do a fresh rebuild on every change.  

Example: https://stackblitz.com/edit/github-6auzzw-4tgbjt?file=app.vue,nuxt.config.ts

**Reply by onpix** [reply]
Wow it works well now! Many thanks!
